scripts/merge-changelogs.ts

```diff
@@ -0,0 +1,203 @@
+import { writeFile, readFile } from 'fs/promises';
+import { GlobSync } from 'glob';
+
+const validMessageTypes = [
+  'Known Issue',
+  'Compatibility Note',
+  'New Functionality',
+  'Improvement',
+  'Fixed Issue',
+  'Updated Dependencies'
+] as const;
+
+interface Message {
+  packageNames: string[];
+  commit: string;
+  type: typeof validMessageTypes[number];
+  version: string;
+  message: string;
+}
+
+interface ContentByVersion {
+  version: string;
+  content: string;
+}
+
+function getPackageName(changelog: string): string {
+  // Get package name without the @sap-cloud-sdk scope
+  const packageNameRegex = /^# @sap-cloud-sdk\/(?<packageName>.+)$/m;
+  const firstLine = changelog.split('\n')[0];
+  const packageNameMatch = firstLine.match(packageNameRegex);
+
+  if (!packageNameMatch?.groups?.packageName) {
+    throw new Error('Changelog should have package name on first line');
+  }
+
+  return packageNameMatch.groups.packageName;
+}
+
+function splitByVersion(changelog: string): ContentByVersion[] {
+  // Explanation: https://regex101.com/r/wCKs2A/3
+  const versionRegex =
+    /## (?<version>[0-9]+\.[0-9]+\..+)(?<content>[^]*?)(?=(\n## |$))/g;
+  const matches = changelog.matchAll(versionRegex);
+  const results: any[] = [];
+  for (const match of matches) {
+    results.push({
+      version: match.groups?.version,
+      content: match.groups?.content.trim()
+    });
+  }
+  return results;
+}
```

Comment:
The split version would break in a number of edge cases. What if there is a changeset like `[Known issue] Don't allow function names with ## in their name`? The regex is not perfect there but less likely to fail.

I prefer the regex over split with lots of `if` checks to make sure the content is as expected. What do you think?

Reply:
True. I still think the split approach is way more readable, how about this: `text.split('\n## ')`

=> Other edge cases you are referring to, I guess, would be something where there is no version or no content? In the regex approach, we would treat this as content, right? How can this happen and do we want to treat it as content? Shouldn't this throw an error anyways?